### PR TITLE
Port Deconflicts for Multi UAV Gazebo Simulations

### PR DESCRIPTION
**Pull request by k2shah**
**Test data / coverage**
Output of diagnostics for a test with 3 UAVs
[3quad_dia.txt](https://github.com/PX4/Firmware/files/2389949/3quad_dia.txt)

**Describe problem solved by the proposed pull request**
Resolves an issue within https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d-posix/rcS#L84-L88 that caused a UDP bind error when spawning multiple UAVs in Gazebo. As a result the UAVs were not able to be armed.  

fixes issue  #10430  

**Describe your preferred solution**
Shifting the UDP port assignments to not have any overlap. The set of ports needed to establish a connection between the simulator and mavros are now a function of the UAV's id and are shifted incrementally for each new UAV. The associated launch file will also reflect these changes. Also fixes an off-by-one mismatch between quad ID's 

**Additional context**
With this fix there's a notable difference between how a single UAV is spawned (hard coded ports within a SDF file) and how the multi_uav launch file works. It might be useful to further consolidate this.

### CODE REVIEW

File: ROMFS/px4fmu_common/init.d-posix/rcS [hunk 1]
```diff
@@ -81,11 +81,11 @@ fi
 
 # multi-instance setup
 # shellcheck disable=SC2154
-param set MAV_SYS_ID $((1+px4_instance))
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d-posix/rcS]: You cannot remove the +1 here, it breaks a lot of things.
If really needed, we can do it, but we need to change more than that.
Did you try what I suggested before in the launch script?
```
<arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
```

File: ROMFS/px4fmu_common/init.d-posix/rcS [hunk 2]
```diff
@@ -81,11 +81,11 @@ fi
...
-simulator_udp_port=$((14560+px4_instance))
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d-posix/rcS]: 14560 needs to stay the same.

File: ROMFS/px4fmu_common/init.d-posix/rcS [hunk 3]
```diff
@@ -81,11 +81,11 @@ fi
...
-udp_offboard_port_local=$((14557+px4_instance))
-udp_offboard_port_remote=$((14540+px4_instance))
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d-posix/rcS]: 14540 also needs to stay. Generally you can change local ports but not remote ones (at least for the first instance).

**Comment by k2shah** [comment]
@bkueng ok I made the changes, I didn't know it was the `tgt_system` that needed to be fixed. The way i have it set up now only 10 UAVs can be spawned, however changes to the documentation can be made to point to this discussion if a user wants to spawn and control more than 10 UAVs? Once this passes jenkin's I can go ahead and make the documentation changes. 

This build seems to work on my end, but I do get some ros warnings about a ID conflict 

`WARN  [mavlink] ignoring CMD with same SYS/COMP (1/1) ID`

File: ROMFS/px4fmu_common/init.d-posix/rcS [hunk 4]
```diff
@@ -81,11 +81,11 @@ fi
 
 # multi-instance setup
 # shellcheck disable=SC2154
-param set MAV_SYS_ID $((1+px4_instance))
+param set MAV_SYS_ID $((px4_instance+1))
 simulator_udp_port=$((14560+px4_instance))
-udp_offboard_port_local=$((14557+px4_instance))
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d-posix/rcS]: This one needs to be different: 14550 is used for a GCS. Maybe 14580?

Comment (k2shah) [reply ROMFS/px4fmu_common/init.d-posix/rcS]: done!

**Comment by bkueng** [comment]
rebased, squashed & merged.

**Pull request closed without merging**